### PR TITLE
Playbook additions and RH-SSO testing setup

### PR DESCRIPTION
**Pull request by briantward**
@itewk @brenton This is still somewhat work in progress.  I haven't ironed out and tested all the manual steps as I was rewriting them to match the changes needed for the playbooks.  However, the playbooks should work fairly well.  Currently tested on 3.9.  I'll give it a roll on 3.11 shortly.

Just looking for open comments and suggestions at the moment. 

Major changes:
- added playbooks to install RH-SSO, Saml-Auth proxy, client, master-config updates, and web console updates
- removed custom image builds and used httpd24 provided in RH Container Catalog (there is also an upstream equivalent)
- created an optional debug image deployment like @itewk's solution but based on httpd24 image sources.
- removed keycloak stuff as it was much older, but we could add back a keycloak specific deployment under a different tag perhaps

### CODE REVIEW

**Comment by brenton** [comment]
I'm definitely impressed.  Let me know how it goes on 3.11.

File: playbooks/install-oauth-on-master.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+- name: Install RequestHeader Oauth Config in master-config.yaml
+  hosts: masters
+  tasks:
+    - name: remove existing HTPasswd provider
```
Comment (itewk) [inline playbooks/install-oauth-on-master.yaml:5]: shouldn't this be an option? I am thinking the case where you are still in troubleshoot mode and using HTPasswd as the way in.

Comment (briantward) [reply playbooks/install-oauth-on-master.yaml:5]: I tried this and found HTPasswd was not an option when using RequestHeader provider (on 3.9).  I'll try again on 3.11 and see if it works there.  Have you gotten it to work?  I know HTPasswd works fine when coupled with the Oauth provider with RH-SSO.

**Comment by briantward** [comment]
> I'm definitely impressed. Let me know how it goes on 3.11.

Playbooks passed on my 3.11 cluster after updating the install oauth playbook to handle containerized master-api.

Still going through manual steps in the README instructions.

File: playbooks/install-oauth-on-master.yaml (hunk 1 above)
Comment (briantward) [reply playbooks/install-oauth-on-master.yaml:5]: Interestingly enough it worked now on 3.11.  Maybe there was a limitation before at some point, so I'll add some code for conditionals and document it.  There is also a revert playbook in the need one has to go back to only HTPasswd

**Pull request closed without merging**